tests/pytorch/nightly/resnet50-mp.libsonnet

```diff
@@ -145,12 +145,80 @@ local tpus = import 'templates/tpus.libsonnet';
     memory: '40Gi',
 
     // Disable XLA metrics report on GPU
-    command+: [
-      '--nometrics_debug',
+    command: [
+      'bash',
+      '-c',
+      |||
+        export PATH=/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+        export LD_LIBRARY_PATH=/usr/local/nvidia/lib:/usr/local/nvidia/lib64
+
+        nvidia-smi
+        nvcc -V
+
+        pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
+        pip install --user https://storage.googleapis.com/pytorch-xla-releases/wheels/cuda/12.1/torch_xla-2.1.0-cp310-cp310-manylinux_2_28_x86_64.whl
+
+        git clone --depth=1 https://github.com/pytorch/pytorch.git
+        cd pytorch
+        git clone https://github.com/pytorch/xla.git
+
+        while true
+        do
+          ip=$(getent hosts ptxla-hello-world-0.headless-svc-$(JOB_NAME) | awk {'print $1'})
+          if [ $? -eq 0 ] && [ \"${ip}\" != \"\" ]
+          then
+            break
+          else
+            sleep 10
+          fi
+        done
+        echo $ip
+
+        PJRT_DEVICE=CUDA torchrun --nnodes=4 --node_rank=$JOB_COMPLETION_INDEX --nproc_per_node=4 --rdzv_endpoint=$ip:12355 xla/test/test_train_mp_imagenet.py  --fake_data --pjrt_distributed --batch_size=128 --num_epochs=1"
+      |||,
     ],
     flags+: {
       modelDir: null,
     },
+
+    jobTemplate+:: {
+      spec+: {
+        completionMode: 'Indexed',
+        completions: 4,
+        parallelism: 4,
+      },
+    },
+
+    podTemplate+:: {
+      spec+: {
+        initContainerMap+:: {
+          'tpu-version': {
+            command: [
+              "kubectl expose headless-svc-$(JOB_NAME) --type='None' --selector='job-name: $(JOB_NAME)'",
+            ],
+            "image": "google/cloud-sdk",
+          },
+        },
+        containerMap+:: {
+          train+: {
+            ports: [
+              {
+                containerPort: 1234,
+              },
+            ],
+          },
+        },
+        subdomain: 'headless-svc-$(JOB_NAME)', # xw32: need to verify.
```

Comment:
hi @will-cromar , currently when I run the test it fails with error `The Job "pt-nightly-resnet50-mp-func-v100-x4-699rm" is invalid: spec.template.spec.subdomain: Invalid value: "headless-svc-$(JOB_NAME)": a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')`. 
The subdomain name (`headless-svc-$(JOB_NAME)`) is the kubernetes service name. Do you know the correct way to make it depends on the job name?